src/SourceBuild/content/build.sh

```diff
@@ -110,169 +162,192 @@ while :; do
-        grep -oP '(?<="'$property_name'": ")[^"]*' "$json_file_path"
-      }
+    "$CLI_ROOT/dotnet" build-server shutdown
 
-      sourceRepository=$(get_property "$releaseManifest" sourceRepository) \
-        || (echo "ERROR: Failed to find sourceRepository in $releaseManifest" && exit 1)
-      sourceVersion=$(get_property "$releaseManifest" sourceVersion) \
-        || (echo "ERROR: Failed to find sourceVersion in $releaseManifest" && exit 1)
+    if [ "$test" == "true" ]; then
+      "$CLI_ROOT/dotnet" msbuild "$scriptroot/build.proj" -bl:"$scriptroot/artifacts/log/$configuration/BuildTests.binlog" -flp:"LogFile=$scriptroot/artifacts/log/$configuration/BuildTests.log" -clp:v=m $properties
+    else
+      "$CLI_ROOT/dotnet" msbuild "$scriptroot/eng/tools/init-build.proj" -bl:"$scriptroot/artifacts/log/$configuration/BuildMSBuildSdkResolver.binlog" -flp:LogFile="$scriptroot/artifacts/log/$configuration/BuildMSBuildSdkResolver.log" /t:ExtractToolPackage,BuildMSBuildSdkResolver $properties
 
-      if [ -z "$sourceRepository" ] || [ -z "$sourceVersion" ]; then
-        echo "ERROR: sourceRepository and sourceVersion must be specified in $releaseManifest"
-        exit 1
-      fi
+      # kill off the MSBuild server so that on future invocations we pick up our custom SDK Resolver
+      "$CLI_ROOT/dotnet" build-server shutdown
+
+      "$CLI_ROOT/dotnet" msbuild "$scriptroot/build.proj" -bl:"$scriptroot/artifacts/log/$configuration/Build.binlog" -flp:"LogFile=$scriptroot/artifacts/log/$configuration/Build.log" $properties
     fi
 
-    # We need to add "fake" .git/ files when not building from a git repository
-    mkdir -p "$GIT_DIR"
-    echo '[remote "origin"]' > "$GIT_DIR/config"
-    echo "url=\"$sourceRepository\"" >> "$GIT_DIR/config"
-    echo "$sourceVersion" > "$GIT_DIR/HEAD"
   fi
-fi
+}
 
-if [ "$CUSTOM_PACKAGES_DIR" != "" ]; then
-  if [ "$runningSmokeTests" == "true" ]; then
-    MSBUILD_ARGUMENTS+=( "-p:CustomSourceBuiltPackagesPath=$CUSTOM_PACKAGES_DIR" )
-  else
-    MSBUILD_ARGUMENTS+=( "-p:CustomPrebuiltSourceBuiltPackagesPath=$CUSTOM_PACKAGES_DIR" )
+if [[ "$clean" == true ]]; then
+  if [ -d "$artifacts_dir" ]; then
+    rm -rf $artifacts_dir
+    echo "Artifacts directory deleted."
   fi
+  exit 0
 fi
 
-if [ -f "${packagesArchiveDir}archiveArtifacts.txt" ]; then
-  ARCHIVE_ERROR=0
-  if [ ! -d "$SCRIPT_ROOT/.dotnet" ] && [ "$CUSTOM_SDK_DIR" == "" ]; then
-    echo "ERROR: SDK not found at '$SCRIPT_ROOT/.dotnet'. Either run prep.sh to acquire one or specify one via the --with-sdk parameter."
-    ARCHIVE_ERROR=1
-  fi
-  if [ ! -f ${packagesArchiveDir}Private.SourceBuilt.Artifacts*.tar.gz ] && [ "$CUSTOM_PACKAGES_DIR" == "" ]; then
-    echo "ERROR: Private.SourceBuilt.Artifacts artifact not found at '$packagesArchiveDir'. Either run prep.sh to acquire it or specify one via the --with-packages parameter."
-    ARCHIVE_ERROR=1
-  fi
-  if [ $ARCHIVE_ERROR == 1 ]; then
-    exit 1
-  fi
-fi
```

Comment:
This is dead code (change that made it so is https://github.com/dotnet/installer/commit/6b37d32420b68a71794ef6007d9f2670799e905f).